### PR TITLE
Support hive `CREATE FUNCTION` syntax

### PR DESCRIPTION
**Pull request by mobuchowski**
Add `CreateFunction` syntax based on Hive: https://cwiki.apache.org/confluence/display/hive/languagemanual+ddl#LanguageManualDDL-Create/Drop/ReloadFunction

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>

### CODE REVIEW

**Review by alamb — APPROVED** [review]
LGTM -- thank you @mobuchowski  

What do you think about conditionalizing this on being part of the hive Dialect?

**Comment by mobuchowski** [comment]
@alamb I think making this conditional would be a good solution. Looking at other DBs that support some form of `CREATE FUNCTION` like [BigQuery](https://cloud.google.com/bigquery/docs/reference/standard-sql/user-defined-functions) or even [MS SQL](https://docs.microsoft.com/en-us/sql/relational-databases/user-defined-functions/create-user-defined-functions-database-engine?view=sql-server-ver15) , the syntax is pretty much unique for each db.

**Comment by alamb** [comment]
This PR appears to have a merge conflict now.

Also, @mobuchowski  are you planning to implement https://github.com/sqlparser-rs/sqlparser-rs/pull/496#issuecomment-1134426353 in this PR, or shall we file a ticket for a follow on task?

**Comment by mobuchowski** [comment]
Sorry, had a busy week. I'll fix the conflict and implement the conditional tomorrow.

**Comment by alamb** [comment]
> Sorry, had a busy week. I'll fix the conflict and implement the conditional tomorrow.



I totally understand!

**Comment by mobuchowski** [comment]
Hey @alamb, fixed the PR and added checking for HiveDialect.

File: tests/sqlparser_hive.rs [hunk 1]
```diff
@@ -232,6 +232,47 @@ fn set_statement_with_minus() {
...
+        }
+        _ => unreachable!(),
+    }
+
+    let generic = TestedDialects {
+        dialects: vec![Box::new(GenericDialect {})],
+    };
+
+    assert_eq!(
+        generic.parse_sql_statements(sql).unwrap_err(),
```
Comment (alamb) [inline tests/sqlparser_hive.rs:263]: 👍

**Review by alamb — APPROVED** [review]
Looks great -- thank you @mobuchowski

**Pull request merged**